<p>Using Spring repository in loop induced unnecessary calculation by the cpu so unless energy consumption.</p>
<h2>Noncompliant Code Example</h2>
<pre>
		private final List&lt;Integer&gt; ids = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);

		List&lt;Employee&gt; employees = new ArrayList&lt;&gt;();
		
		for (Integer id: ids) {
            Optional&lt;Employee&gt; employee = employeeRepository.findById(id); // Noncompliant
            if (employee.isPresent()) {
                employees.add(employee.get());
            }
        }

</pre>
<h2>Compliant Solution</h2>
<pre>
		private final List&lt;Integer&gt; ids = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
		List&lt;Employee&gt; employees = employeeRepository.findAllById(ids);
</pre>
