<p>Using System.arraycopy to copy arrays</p>
<p>
	Programs spend most of the time in loops. These can be resource consuming, especially when they integrate heavy processing (IO access). Moreover, the size of the data and processing inside the loops will not allow full use of hardware mechanisms such as the cache or compiler optimization mechanisms.<br />
	For example, an array copy is potentially a non-performance source if it is poorly designed. Indeed, the use of a single copy loop can be twice as consuming as dedicated methods.<br />
	Loops must be optimized to reduce processing time and make full use of hardware and processor mechanisms and optimizations.<br />
	In the case of table copying (table), the native System.arraycopy.<br />
	We can also use copyOf or clone that are slightly less efficient.<br />
	The looping method will be outlawed.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
	int len = array.length;
	boolean[] copy = new boolean[array.length];
	for (int i = 0; i < len; i++) {
  		copy[i] = array[i];
	}
	return copy;
</pre>
<h2>Compliant Solution</h2>
<pre>
	int[] copy = new int[array.length];
	System.arraycopy(array, 0, copy, 0, array.length);
	return copy;
</pre>
