<p>
    If you know in advance how many characters would be appended, initialize builder/buffer with the appropriate size.
    They will thus never have to be resized, It saves CPU cycles so unless energy consumption.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    StringBuilder sb = new StringBuilder(); // Noncompliant
    for (int i = 0; i < 100; i++) {
       sb.append(...);
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
    StringBuilder sb = new StringBuilder(100);
    for (int i = 0; i < 100; i++) {
       sb.append(...);
    }
</pre>
