<p>Optimize read file exception</p>
<h2>Noncompliant Code Example</h2>
<pre>
		public void readPreferences(String filename) {
		  //...
		  InputStream in = null;
		  try {
			in = new FileInputStream(filename);
		  } catch (FileNotFoundException e) {
			logger.log(e);
		  }
		  in.read(...);
		  //...
		}

</pre>
<h2>Compliant Solution</h2>
<pre>
		public void readPreferences(String filename)
			throws IllegalArgumentException,
				   FileNotFoundException, IOException {
		  if (filename == null) {
			throw new IllegalArgumentException ("filename is null");
		  }  //if
		  //...
		  InputStream in = new FileInputStream(filename);
		  //...
		}
</pre>
