<p>While iterating through any collection get the size of the collection beforehand and never get it during iteration, It saves CPU cycles so unless energy consumption. The sample is provided below as an illustration which is to be avoided as follows.</p>
<h2>Noncompliant Code Example</h2>
<pre>
		List&lt;String&gt; objList = getData();

        for (int i = 0; i < objList.size(); i++) {
            // execute code
        }

</pre>
<h2>Compliant Solution</h2>
<pre>
        List&lt;String&gt; objList = getData();

        int size = objList.size();
        for (int i = 0; i < size; i++) {
            // execute code
        }
</pre>
