<p>Avoid the use of Foreach with Arrays</p>
<h2>Noncompliant Code Example</h2>
<pre>
		private final Integer[] intArray = new Integer[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };

		for (Integer i : intArray) {
			...
		}

</pre>
<h2>Compliant Solution</h2>
<pre>
		private final List<Integer> intList = Arrays.asList(new Integer[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });

		for (Integer i : intList) {
			...
		}
</pre>
