<p>Using List instead of Arrays with Foreach save CPU cycles calculations and RAM consumption.</p>
<h2>Noncompliant Code Example</h2>
<pre>
		private final Integer[] intArray = new Integer[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };

		for (Integer i : intArray) {
			...
		}

</pre>
<h2>Compliant Solution</h2>
<pre>
		private final List&lt;Integer&gt; intList = Arrays.asList(new Integer[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });

		for (Integer i : intList) {
			...
		}
</pre>
