<p>Do not unnecessarily assign values to variables</p>
<h2>Noncompliant Code Example</h2>
<pre>
String var1 = getValue();
return var1;

String var2 = "hello"
var2 = "world"        //Non compliant cause never assigned 

</pre>
<h2>Compliant Solution</h2>
<pre>return getValue();</pre>
