<p>
    Avoid using Pattern.compile() in a non-static context.
    This operation requires a non negligible amount of computational power, Using a single match saves CPU cycles and RAM consumption.
</p>

<h2>Noncompliant Code Example</h2>
<pre>

    public class AvoidRegexPatternNotStatic {

        public boolean foo() {
            final Pattern pattern = Pattern.compile("foo"); // Noncompliant
            return pattern.matcher("foo").find();
        }

    }

</pre>

<h2>Compliant Solution N°1</h2>
<pre>

    public class ValidRegexPattern {

        private static final Pattern pattern = Pattern.compile("foo"); // Compliant

        public boolean foo() {
            return pattern.matcher("foo").find();
        }

    }

</pre>

<h2>Compliant Solution N°2</h2>
<pre>

    public class ValidRegexPattern2 {

        private final Pattern pattern = Pattern.compile("foo"); // Compliant

        public boolean foo() {
            return pattern.matcher("foo").find();
        }

    }

</pre>

<h2>Compliant Solution N°3</h2>
<pre>

    public class ValidRegexPattern3 {

        private final Pattern pattern;

        public ValidRegexPattern3() {
            pattern = Pattern.compile("foo"); // Compliant
        }

        public boolean foo() {
            return pattern.matcher("foo").find();
        }

    }

</pre>
