<p>Don't set const parameter in batch update => Put its in query. Creating this parameter and destroying it consumes CPU cycles and RAM unnecessarily.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public void foo() {
    	...
    	String query = "insert into mytable values(?,?,?)";
        ...
        for(DummyClass o : list) {
			stmt.setInt(1, 123);
			stmt.setString(2, o.getName());
			stmt.setDouble(3, o.getPrice());
			stmt.addBatch();
		}
        ...
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
    public void foo() {
    	...
    	String query = "insert into mytable values(123,?,?)";
        ...
        for(DummyClass o : list) {
			stmt.setString(1, o.getName());
			stmt.setDouble(2, o.getPrice());
			stmt.addBatch();
		}
        ...
    }
</pre>
