<p>try-with-resources Statement needs to be implemented for any object that implements the AutoCloseable interface, it save computer resources.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    private static void printFileJava7() throws IOException {
        FileInputStream input = new FileInputStream("file.txt");
        int data = input.read();
        while(data != -1){
            System.out.print((char) data);
            data = input.read();
        }
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
    private static void printFileJava7() throws IOException {
        try(FileInputStream input = new FileInputStream("file.txt")) {
            int data = input.read();
            while(data != -1){
                System.out.print((char) data);
                data = input.read();
            }
        }
    }
</pre>
