<p>Use PreparedStatement instead of Statement, that's because SQL will validate the query only once, whereas if you just use a statement it will validate the query each time.
    This induced unnecessary calculation by the cpu so unless energy consumption.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public void select() {
        Statement statement = connection.createStatement();
        statement.executeUpdate("INSERT INTO persons(id, name) VALUES(2, 'Toto')");  // Noncompliant
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
    public void select() {
        PreparedStatement statement = connection.prepareStatement(INSERT INTO persons(id, name) VALUES(?, ?));

        statement.setInt(1, 2);
        statement.setString(2, "Toto");
        statement.executeQuery();
    }
</pre>
