<p>Use of methods for basic operations</p>
<h2>Noncompliant Code Example</h2>
<pre>
    $min = min($a, $b); // NOK
</pre>
<h2>Compliant Solution</h2>
<pre>
    $min = ($a < $b) ? $a : $b; // NOK
</pre>
