<p>Do not call a function when declaring a for-type loop in order to avoid function calls each iterations. It saves CPU cycles.</p>
<h2>Noncompliant Code Example</h2>
<pre>
for ($i = 0; $i <= foo(); $i++) { // NOK
  ......
}
</pre>
<h2>Compliant Solution</h2>
<pre>
$maxI = foo();
for ($i = 0; $i <= $maxI; $i++) {
  .....
}

  OR

for ($i = 0, $maxI = foo(); $i <= $maxI; $i++) {
  .....
}
}</pre>
