<p>Databases servers have to solves fileds regarding to schema. Knowing and using the schema save CPU cycles and network transfer.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public function foo() {
        ...
        $baseQuery = "SELECT * FROM users"; // Noncompliant

        ...
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
    public function foo() {
        ...
        $baseQuery = "SELECT id,name, adress FROM users ";
        ...
   }
</pre>
