<p>Avoid Getters and Setters in class, It increase unless RAM memory usage.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class Client():

    def __init__(self, age):
        self.age = age

    def get_age(self):
        return self.age

    def set_age(self, age):
        self.age = age

client = Client(25)
client.get_age() # Getter inutile
client.set_age(25) # Setter inutile

</pre>
<h2>Compliant Solution</h2>
<pre>
class Client():

    def __init__(self, age):
        self.age = age

    def get_age(self):
        return self.age

    def set_age(self, age):
        self.age = age

client = Client(25)
client.age # RÃ©cupÃ©rer l'attribut age
client.age = 26 # Modifier l'attribut age
</pre>
