<p>Do not call a function when declaring a for-type loop in order to avoid function calls each iterations. It saves CPU cycles.</p>
<h2>Noncompliant Code Example</h2>
<pre>
for i in my_function():
    ......

</pre>
<h2>Compliant Solution</h2>
<pre>
limit = my_function()
for i in limit:
    ......
</pre>
